cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -730,7 +727,8 @@ func (p *BicepProvider) Destroy(
 	options provisioning.DestroyOptions,
 ) (*provisioning.DestroyResult, error) {
 	modulePath := p.modulePath()
-	// TODO: Report progress, "Compiling Bicep template"
+	p.console.ShowSpinner(ctx, "Discovering resources to delete...", input.Step)
+	defer p.console.StopSpinner(ctx, "", input.StepDone)
```

Comment:
Just checking - in the common case we'll call `StopSpinner` twice now, once on L793 and then once when this function returns.  That's okay right?

Reply:
The pattern to call `StopSpinner` multiple times is generally safe, per the docs:

```
	// Stop the current spinner from the console and change the spinner bar for the lastMessage
	// Set lastMessage to empty string to clear the spinner message instead of a displaying a last message
	// If there is no spinner running, this is a no-op function
```

So calling `StopSpinner` multiple times is ok.

But the order of operations here may be interesting -- I'm curious if @vhvb1989 views the setup here as idiomatic, i.e. 

In the happy path:
1. In the start of `Destroy()`, we create a spinner that 
2. The happy path spinner stops on L793
3. In the end of `Destroy()` we stop a spinner again

Supposedly, if we create another spinner in-between L793 and the end of the function that perhaps creates another spinner this may slightly quirky -- but if we did add another StopSpinner to the defer stack, I think we do observe the overall correct behavior. So from a correctness standpoint, I think we're solid, however I'm not sure if we like it idiomatically.

Reply:
seems fine to me.